packages/x-date-pickers/src/LocalizationProvider/LocalizationProvider.tsx

```diff
@@ -33,13 +38,27 @@ export interface LocalizationProviderProps {
   dateLibInstance?: any;
   /** Locale for the date library you are using */
   locale?: string | object;
+  /**
+   * Locale for components texts
+   */
+  localeText?: Partial<PickersLocaleText>;
 }
 
 /**
  * @ignore - do not document.
  */
-export function LocalizationProvider(props: LocalizationProviderProps) {
-  const { children, dateAdapter: Utils, dateFormats, dateLibInstance, locale } = props;
+export function LocalizationProvider(inProps: LocalizationProviderProps) {
+  const props = useThemeProps({ props: inProps, name: 'MuiLocalizationProvider' });
+
+  const {
+    children,
+    dateAdapter: Utils,
+    dateFormats,
+    dateLibInstance,
+    locale,
+    localeText = {},
```

Comment:
If you define the default value here, then `contextValue` memoization does not work when there is no locale.

```ts
        ...enUS.components.MuiLocalizationProvider.defaultProps.localeText,
        ...localeText ?? {},
```

That would be better I think